src/libsyntax/ext/expand.rs

```diff
@@ -539,6 +541,45 @@ impl<'a, 'b> MacroExpander<'a, 'b> {
         }
     }
 
+    fn extract_proc_macro_attr_input(&self, tokens: TokenStream, span: Span) -> TokenStream {
+        match tokens.trees().next() {
+            Some(TokenTree::Delimited(_, delim)) => delim.tts.into(),
```

Comment:
This still accepts `#[attr(delimited token stream) + anything else]`.
Could you add a test making sure that the example above and also `#[attr = "something"]` are rejected.

Reply:
Aha excellent point, tests added and fixed